### PR TITLE
Split `REnv` into Global and Local Envs

### PR DESCRIPTION
**Pull request by ranjitjhala**
To enable printing of just the "local" part, the global one being the binders that correspond to module imports.

### CODE REVIEW

File: tests/test.hs [hunk 1]
```diff
@@ -158,7 +158,7 @@ knownToFail Z3   = [ "tests/pos/linspace.hs" ]
 testCmd :: FilePath -> FilePath -> FilePath -> SmtSolver -> LiquidOpts -> String
 ---------------------------------------------------------------------------
 testCmd bin dir file smt (LO opts)
-  = printf "cd %s && %s --verbose --smtsolver %s %s %s" dir bin (show smt) file opts
+  = printf "cd %s && %s --smtsolver %s %s %s" dir bin (show smt) file opts
```
Comment (gridaphobe) [inline tests/test.hs]: please don't remove the verbose output, it can be very useful on travis!

Comment (ranjitjhala) [inline tests/test.hs]: has it ever been?

Comment (ranjitjhala) [inline tests/test.hs]: Plus:

+ Don't want to have the numbers generated by `stack/cabal test` to be corrupted by debugging, as they are also used to measure performance over time,

+ We can always add the `--verbose` as an extra parameter to `stack/cabal test` if we really want to generate the verbose logs on travis, no need to hardwire the thing inside the script...

Comment (gridaphobe) [inline tests/test.hs]: Fair enough, let's just add `--verbose` to the travis scripts then. It has been invaluable on the (rare) occasion that I can't reproduce the failure locally.

**Comment by ranjitjhala** [comment]
Some more changes, to *avoid* throwing the VV into the environment (and hence decrease the number of non-linear KVars).

**Comment by ranjitjhala** [comment]
merging because travis is ok with it...

**Pull request merged**